dev/io.openliberty.org.apache.myfaces.4.0_fat/fat/src/io/openliberty/org/apache/myfaces40/fat/tests/SubscribeToEventTest.java

```diff
@@ -0,0 +1,78 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.org.apache.myfaces40.fat.tests;
+
+import static org.junit.Assert.assertEquals;
+
+import java.net.URL;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.rules.TestName;
+import org.junit.runner.RunWith;
+
+import com.gargoylesoftware.htmlunit.WebClient;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+import com.ibm.websphere.simplicity.ShrinkHelper;
+
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.HttpUtils;
+import io.openliberty.org.apache.myfaces40.fat.FATSuite;
+
+/**
+ * Faces 4.0 modified UIComponent#subscribeToEvent() and UIComponent#getListenersForEventClass()
+ * APIs to be more developer friendly.
+ * This test verifies the API changes where correctly implemented.
+ */
+@RunWith(FATRunner.class)
+public class SubscribeToEventTest {
+
+    private static final String APP_NAME = "SubscribeToEventTest";
+
+    @Server("faces40_subscribeToEventTest")
+    public static LibertyServer server;
+
+    @Rule
+    public TestName name = new TestName();;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        ShrinkHelper.defaultDropinApp(server, APP_NAME + ".war", "io.openliberty.org.apache.faces40.fat.subscribe");
+
+        server.startServer(SimpleTest.class.getSimpleName() + ".log");
+    }
+
+    @AfterClass
+    public static void cleanUp() throws Exception {
+        // Stop the server
+        if (server != null && server.isStarted()) {
+            server.stopServer();
+        }
+    }
+
+    /**
+     * FIXME - this test currently fails because a single subscribe causes duplicate values to be returned from list.
+     * Opened PR with MyFaces to fix this behavior: https://github.com/apache/myfaces/pull/509
+     */
+//    @Test
```

Comment:
Fix was merged in, so this test can be un-commented

Reply:
https://jakarta.ee/specifications/faces/4.0/apidocs/jakarta/faces/component/uicomponentbase#unsubscribeFromEvent(java.lang.Class,jakarta.faces.event.ComponentSystemEventListener)

```
Remove the listener instance referenced by argument componentListener as a listener for events of type eventClass originating from this specific instance of UIComponent. When doing the comparison to determine if an existing listener is equal to the argument componentListener (and thus must be removed), the equals() method on the existing listener must be invoked, passing the argument componentListener, rather than the other way around. 
```

It looks like you should use the existing MyTestListener?